src/nvim/ops.c

```diff
@@ -1391,6 +1391,7 @@ int insert_reg(int regname, yankreg_T *reg, bool literally_arg)
           // Insert a newline between lines and after last line if
           // y_type is kMTLineWise.
           if (reg->y_type == kMTLineWise || i < reg->y_size - 1) {
+            // likely the place to insert a shift+NL in prompt buffers figure out how to insert it (shadman)
```

Comment:
?

Reply:
This is basically how inserting register is supposed to work in insert mode with `<C-r>`. Currently, it just raw inserts the reg. As a result, if the reg has multiple lines, it just submits the prompt. I think it makes sense to paste the reg completely like `p` is normal mode. For that, we need to insert `<S-CR>` in place of `<CR>`  here on the prompt buffers. Any idea how to do so? I had asked on the dev channel as well as justin on element but didn't get an answer. I thought you or bfredl would be the best person to ask on these stuff but didn't yet got around to it. I had left the comment as a marker so I can find the location later :]

Also another thing is do we want `<C-r>` to be raw input? In that case keeping the current behavior is fine.

Reply:
I mean, why is the comment here?

Reply:
It was a temporary comment. To mark where change needs to be made. So I can find it later without having to read most of the logic there again. I'm going to remove it before merge.